scalableminds/webknossos
# Clean up frontend deprecations and warnings

Lots of react deprecation warnings and html structure warnings were clogging up the console, making it hard to see relevant information.
Most of the `componentWillMount`, `componentWillUpdate`, and `componentWillReceiveProps` occurrences could be replaced with the `didMount`/`didUpdate` alternatives fairly easily. However, for some of them I didn't see a good way.
The `// eslint-disable-next-line react/no-did-update-set-state` are a little annoying since having the `setState` behind an if-comparison-clause seems to be the officially recommended way :shrug: 

### URL of deployed dev instance (used for testing):
- https://cleanupdeprecationsandwarnings.webknossos.xyz

### Steps to test:
- Test that the persistence still works by entering a search term in one of the admin views and then reloading the page
- Test the other components that were changed non-trivially: permissions_and_teams_modal_view, team_selection_component, vector_input, merge_modal_view, editable_text_label, histogram_view
- While clicking around (almost) no deprecation warnings should turn up in the console (except for Remarkable which already is the newest version, but doesn't seem to be maintained any longer :/)

### Issues:
- fixes clogged up console

------
(Please delete unneeded items, merge only when none are left open)
